### PR TITLE
fix: move slack channel to top of table description

### PR DESCRIPTION
**Pull request by zhangvi7**
### Minor Querybook Improvements for Tier 1 tables
- Pull up slack channel to the top of the table description / highlight it. 
#### Test Plan

Bring slack channel to top of table details and add channel link

### CODE REVIEW

File: querybook/webapp/components/DataTableViewOverview/DataTableViewOverview.tsx [hunk 1]
```diff
@@ -82,6 +82,24 @@ function useRefreshMetastore(table: IDataTable) {
     return [handleRefreshTable, isRefreshing] as const;
 }
 
+function createChannelLinkDOM(customProperties) {
+    const channelsKey = 'channels';
+    const channelsValue = customProperties[channelsKey]?.toString() ?? '';
+    const channelsName = channelsValue.slice(1);
+    const channelLink = `https://pinterest.slack.com/app_redirect?channel=${channelsName}`;
```
Comment (jczhong84) [inline querybook/webapp/components/DataTableViewOverview/DataTableViewOverview.tsx:89]: this is too Pinterest specific for the opensource repo and need a better solution for it. let's not add a link for now. maybe can add link in the properties when syncing from pincat

Comment (zhangvi7) [reply querybook/webapp/components/DataTableViewOverview/DataTableViewOverview.tsx]: removed link for now

**Review by jczhong84 — DISMISSED** [review]

File: querybook/webapp/components/DataTableViewOverview/DataTableViewOverview.tsx [hunk 2]
```diff
@@ -82,6 +82,20 @@ function useRefreshMetastore(table: IDataTable) {
...
+    return channelsValue === '' ? null : (
+        <KeyContentDisplay
+            key={channelsKey}
+            keyString={titleize(channelsKey, '_', ' ')}
+        >
+            {channelsValue}
+        </KeyContentDisplay>
+    );
+}
+
```
Comment (czgu) [inline querybook/webapp/components/DataTableViewOverview/DataTableViewOverview.tsx:98]: this is duplicating a lot of code specially for channels, could you have something like:

```
/**
* any custom properties in this array will be shown on top following the order of the array
*/
const pinnedCustomProperties = ['channels'];
```
then when you go over customPropertiesDOM, make orderedProperties show at the beginning

**Review by czgu — CHANGES_REQUESTED** [review]

File: querybook/webapp/components/DataTableViewOverview/DataTableViewOverview.tsx (hunk 2 above)
Comment (zhangvi7) [reply querybook/webapp/components/DataTableViewOverview/DataTableViewOverview.tsx]: - Refactored KeyContentDisplay to helper 
- Added `pinnedCustomProperties` to displayed pinned custom properties at top and unpinned custom properties at bottom
- Updated test plan with screenshots

File: querybook/webapp/components/DataTableViewOverview/DataTableViewOverview.tsx [hunk 3]
```diff
@@ -151,22 +171,22 @@ export const DataTableViewOverview: React.FC<
...
-                    <Link to={valueStr} newTab>
-                        {valueStr}
-                    </Link>
-                ) : (
-                    valueStr
-                )}
-            </KeyContentDisplay>
-        );
-    });
+    const customProperties = Object.entries(table.custom_properties ?? {});
```
Comment (czgu) [inline querybook/webapp/components/DataTableViewOverview/DataTableViewOverview.tsx:174]:
```ts
const customProperties = table.custom_properties;
const pinnedPropertiesDOM = pinnedCustomProperties.filter(p => p in customProperties).map((key) => {
    const value = customProperties[key];
    return ...
};

const otherPropertiesDOM = Object.entries(customProperties).filter(([key]) => !pinnedCustomProperties.includes(key)).map(...);
```

File: querybook/webapp/components/DataTableViewOverview/DataTableViewOverview.tsx [hunk 4]
```diff
@@ -56,6 +56,26 @@ const dataTableDetailsRows = [
     'column_count',
 ];
 
+/**
+ * Any custom properties in this array will be shown on top following the order of the array
+ */
+const pinnedCustomProperties = ['channels'];
+
+function createKeyContentDisplayElement(key: string, value: string | number) {
```
Comment (czgu) [inline querybook/webapp/components/DataTableViewOverview/DataTableViewOverview.tsx:64]: make this into a React Component?

**Comment by zhangvi7** [comment]
- moved to react component
- address comment

**Review by czgu — COMMENTED** [review]
LGTM 👏 , let's fix the errors before r+

**Review by czgu — APPROVED** [review]

**Pull request merged**